### DISCUSSION TITLE
Support for multiple flutter apps with single serverpod server

### DISCUSSION THREAD

**Discussion by DigiReaper** [serverpod/serverpod#2452] [Q&A] [answered]
as mentioned, is it possible to use multiple flutter apps with a single server.

**Comment by SandPod** [comment, marked as answer]
@DigiReaper could you give a little more information about the setup you are envisioning? What is your use-case?

It shouldn't be a problem to connect multiple Flutter applications to a single Serverpod server, you can import the same client to multiple projects.

(a bit unrelated) @lukehutch I think you have missed the possibility of creating custom overrides for the core authentication in Serverpod documented here: https://docs.serverpod.dev/concepts/authentication/custom-overrides.

The authentication module we provide is opinionated, but you can easily implement your own if it is desired.

**Reply by DigiReaper** [reply]
hey @SandPod 
let me briefly explain what i'm trying to do.
I'm working on a inventory management application which has different screens based on the user type( admin, supervisor, etc)
so i wanna have a single serverpod server and multiple flutter apps since they all need to access the same database and some functions.

while we're at it, is it possible to create users from server code. i have a use case where the admin should be able to create user credentials for employess and such.

thanks in advnace.

**Reply by lukehutch** [reply]
OK, then I misunderstood what you were trying to do! You can ignore my comment, it's not relevant.

You can definitely have different user types in the same system. You can protect their endpoints using the "roles" system (see the docs). And your Flutter app can show a different UI based on what a user's roles are, after they are logged in.

**Reply by lukehutch** [reply]
@DigiReaper re creating users in code: see my comment here: https://github.com/serverpod/serverpod/issues/2445#issuecomment-2215509308
